.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,46 @@
+name: Nightly integration tests
+
+# Run on request and every day at 3 AM UTC
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 0 3 * * *
+
+jobs:
+  quantinuum:
+    name: Quantinuum (simulator)
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/${{ github.repository }}:latest
+      options: --user root
+
+    steps:
+      - name: Get code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Set git safe
+        run: git config --global --add safe.directory $PWD
+
+      - name: Install requirements
+        run: |
+          apt-get update
+          apt-get install -y curl jq
+
+      - name: Setup quantinum account
+        run: |
+          curl -X POST -H "Content Type: application/json" -d '{ "email":"${{ secrets.BACKEND_LOGIN_EMAIL }}","password":"${{ secrets.QUANTINUUM_PASSWORD }}" }' https://qapi.quantinuum.com/v1/login > credentials.json
+          id_token=`cat credentials.json | jq -r '."id-token"'`
+          refresh_token=`cat credentials.json | jq -r '."refresh-token"'`
+          echo "key: $id_token" > ~/.quantinuum_config
+          echo "refresh: $refresh_token" >> ~/quantinuum_config
+
+      - name: Build and run
+        run: |
+          for filename in test/NVQPP/integration/*.cpp; do
```

Comment:
We just merged in some hardware examples that aren't tested by the CI yet ([docs/sphinx/examples/*/providers](https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/examples/cpp/providers/quantinuum.cpp)). It might be worth testing that single quantinuum file and, now or eventually, running the [python](https://github.com/NVIDIA/cuda-quantum/blob/main/docs/sphinx/examples/python/providers/quantinuum.py) example as well

Reply:
Thanks for bringing it up. I think it would be better to leave python tests for a future PR, tho. First, let's see how this workflow will work, next I will add IONQ backend, and then another workflow where we install wheel in a clean environment and test providers. What you think?

Reply:
Sounds good to me 👍